### PR TITLE
WT-12750 Remove LSM test code

### PR DESCRIPTION
**Pull request by lukech**
This PR aims to remove all LSM-related test code in the WiredTiger code base, as the first step in deprecating LSM support (as part of [SPM-3441](https://jira.mongodb.org/browse/SPM-3441)).

Commits are broken down by testing framework or functionality for reviewing convenience.

There are lots of changes in this PR (I'm sorry). I'd appreciate it if 
- @sueloverso could help review test/checkpoint and wtperf (possibly test/format too) changes.
- @jeremythorp to cover evergreen and code coverage changes.
- @Sean04 and @luke-pearson to have a general review of all changes.

### CODE REVIEW

File: bench/analytics/atlas/charts/Perf.charts
Comment (lukech) [inline bench/analytics/atlas/charts/Perf.charts:1]: @Siddhartha8899, please help confirm if it's okay to remove this file. It looks like the file content is not being kept up-to-date anyway.

File: bench/analytics/custom_analysis/atlas_analysis.ipynb
Comment (lukech) [inline bench/analytics/custom_analysis/atlas_analysis.ipynb:24]: @ajmorton We chatted about renaming the `test_name` offline, and here is the change.

File: bench/wtperf/runners/overflow-10k.wtperf
Comment (lukech) [inline bench/wtperf/runners/overflow-10k.wtperf:1]: Changes in this file are made by referring to the corresponding settings in `overflow-130k.wtperf`.

File: test/evergreen/code_coverage/test_table.json
Comment (lukech) [inline test/evergreen/code_coverage/test_table.json:5]: The below config section was removed if you have issue telling it apart by flesh eyes (just like me :))
```
lsm=(auto_throttle=true,bloom=true,bloom_bit_count=16,bloom_config=,bloom_hash_count=8,bloom_oldest=false,chunk_count_limit=0,chunk_max=5GB,chunk_size=10MB,merge_custom=(prefix=,start_generation=0,suffix=),merge_max=15,merge_min=0),
```

File: test/fuzz/config/corpus/corpus-10
Comment (lukech) [inline test/fuzz/config/corpus/corpus-10:1]: I don't think test/fuzz is integrated into our Evergreen CI system yet, and assume removing a test config is safe.

File: test/suite/test_backup03.py
Comment (lukech) [inline test/suite/test_backup03.py:1]: This test is solely written for LSM.

File: test/suite/test_bug013.py
Comment (lukech) [inline test/suite/test_bug013.py:1]: This test is solely written for LSM.

File: test/suite/test_drop02.py
Comment (lukech) [inline test/suite/test_drop02.py:1]: This test was solely written for LSM and will be removed.

File: test/suite/test_schema03.py
Comment (lukech) [inline test/suite/test_schema03.py:277]: It's not obvious to me if there's a chance the ratio adjustment made here would impact the testing effectiveness. Appreciate any pair of eyes.

File: test/format/format_config.c
Comment (etienneptl) [inline test/format/format_config.c:244]: Removing this will fire our FIXME check for WT-4162.

Comment (etienneptl) [reply test/format/format_config.c:244]: My bad, I misread.

**Review by jeremythorp — APPROVED** [review]
Hi @lukech , I've reviewed your PR with (as requested) a particular focus on code coverage and evergreen changes, and the changes LGTM!

**Review by luke-pearson — COMMENTED** [review]
WT-8867 Will need closing out after this is merged.

File: test/suite/test_schema03.py
Comment (luke-pearson) [reply test/suite/test_schema03.py:277]: To me this looks okay

**Comment by lukech** [comment]
> [WT-8867](https://jira.mongodb.org/browse/WT-8867) Will need closing out after this is merged.
> [WT-3682](https://jira.mongodb.org/browse/WT-3682) also needs closing out post merge.

Yep, the closure of the 2 tickets (together with other LSM-related tickets) will be tracked by a follow-on ticket WT-12752.

**Review by Sean04 — APPROVED** [review]
LGTM, thank you for this, these are a lot of changes!

File: test/checkpoint/test_checkpoint.h
Comment (sueloverso) [inline test/checkpoint/test_checkpoint.h:45]: Does this comment need updating if we removed on of the table types? I read the comment and it actually isn't clear. It almost feels like it wasn't updated in the past when it went to 4 table types and has been wrong. But more eyes should look at it.

File: test/checkpoint/test_checkpoint.h
Comment (sueloverso) [inline test/checkpoint/test_checkpoint.h:47]: Does this need to be reduced by one now? The max enum now I think will be 3.

File: test/csuite/timestamp_abort/main.c
Comment (sueloverso) [inline test/csuite/timestamp_abort/main.c:1489]: Not in this file - but the `-L` argument is used in `timestamp_abort/smoke.sh` so those lines need to be removed from there too.

**Review by sueloverso — APPROVED** [review]
I have a couple small comments. You might double check any other `smoke.sh` files in other directories for a similar issue. But I don't need to review again after those are fixed. LGTM after minor changes.

File: bench/wtperf/wtperf_opt_inline.h
Comment (lukech) [reply bench/wtperf/wtperf_opt_inline.h:98]: The `compact` option is currently used in some btree "file" type configurations. 
```
$ grep compact bench/wtperf/runners/*
bench/wtperf/runners/500m-btree-populate-large.wtperf:compact=true
bench/wtperf/runners/500m-btree-populate.wtperf:compact=true
bench/wtperf/runners/overflow-10k.wtperf:compact=true
bench/wtperf/runners/overflow-130k-large.wtperf:compact=true
bench/wtperf/runners/overflow-130k.wtperf:compact=true
```

File: test/checkpoint/test_checkpoint.h
Comment (lukech) [reply test/checkpoint/test_checkpoint.h:47]: Good catch.

File: test/checkpoint/test_checkpoint.h
Comment (lukech) [reply test/checkpoint/test_checkpoint.h:45]: Though I have no idea what "special type of mixed" (i.e. `MIX`) means, it does seem [when the 4th type was introduced](https://github.com/wiredtiger/wiredtiger/commit/9b7021497f4c61dca5448778b5d78c1990c6a561#diff-23188d7b317796883e7b7d2350e4fa9c45c82baaef7957189c1fe1a4a2170abf) the comment was not updated. So now we are removing one type and it offsets the missing update :)

File: test/csuite/timestamp_abort/main.c
Comment (lukech) [reply test/csuite/timestamp_abort/main.c:1489]: Thanks for pointing it out. The 2 command lines in `timestamp_abort/smoke.sh` that make use of the "-L" arg are currently commented out. I'll remove them.

Comment (lukech) [reply test/csuite/timestamp_abort/main.c:1489]: See [7915cc6](https://github.com/wiredtiger/wiredtiger/pull/11278/commits/7915cc61f3d937a92907c9ab7892b8e52b3fc448)

File: test/checkpoint/test_checkpoint.h
Comment (lukech) [reply test/checkpoint/test_checkpoint.h]: See [7915cc6](https://github.com/wiredtiger/wiredtiger/pull/11278/commits/7915cc61f3d937a92907c9ab7892b8e52b3fc448)

**Pull request merged**
